config/routes.rb

```diff
@@ -263,11 +263,9 @@
     end
   end
 
-  resources :audits, :only => [:index] do
-    collection do
-      get 'auto_complete_search'
-    end
-  end
+  match '/audits/auto_complete_search' => 'audits#auto_complete_search', :via => [:get]
+  resources :audits, :only => [:index, :show], constraints: ->(req) { req.format == :json }
```

Comment:
@ohadlevy Had to put this line after the auto_complete match for everything to work. still 3 lines needed.

Reply:
there is no show, so imho you can just ```only: :index```

Reply:
updated